go/process/health_dao.go

```diff
@@ -147,10 +167,18 @@ func expireAvailableNodes() error {
 			`,
 		registrationPollSeconds*2,
 	)
-	return log.Errore(err)
+	if err != nil {
+		log.Errorf("ExpireAvailableNodes: failed to remove old entries: %+v", err)
+	}
+
+	// show we're done
+	expiryMutex.Lock()
+	expiryRunning = false
+	expiryMutex.Unlock()
```

Comment:
While the lock/expiryRunning mechanism is correct, I'm curious why your are so concerned with this? With this PR, this code will only run once per minute. A couple dozen other similar "cleanup" functions similarly run once per minute, and do not have this protection.

If this function blocks for the duration of one minute, there are greater problems, and skipping the next run is _nice_ but wont be the thing to solve those greater problems. I'd suggest removing this protection code.